pubtools/_quay/iib_operations.py

```diff
@@ -137,12 +139,14 @@ def task_iib_add_bundles(
         quay_user=target_settings["dest_quay_user"],
         quay_password=target_settings["dest_quay_password"],
     )
+    cert, key = pm.hook.get_cert_key_paths_plugin(server_url=target_settings["pyxis_server"])
```

Comment:
if pubtools-quay is running without pub, this should return None, if it's not registered right? Or am I missing something? In that case we should add fallback to target_settings

Reply:
A couple of things here:

1) Not sure if this actually works.  Was it realistically tested? The hook call doesn't directly return the result, it returns a list of results, since there can be multiple hookimpl attached.  See https://pluggy.readthedocs.io/en/latest/#collecting-results and also see "firstresult" if you want to use that to handle the "more than one hookimpl" scenario.

See also my later comment in a test where pm was mocked.

2) I agree that there should be some fallback or way of making this work outside of Pub.

Actually this usage of hooks doesn't match what I had in mind when introducing the system originally, and I'm having trouble forming an opinion whether this is a good idea.  But I'm confident at least that it shouldn't be the *only* way this code can be reasonably used because that goes against some of the goals of splitting our projects up.

It's intended that each of the projects like pubtools-quay etc should be usable standalone, not only inside of Pub. 
 If we set things up like "this code *requires* a hook to return a value, and the only implementer lives in Pub" then that's pushing us in the opposite direction - we're creating more reasons that this code can't possibly work unless you run it in Pub.

I don't know exactly what the default behavior should be, but it should be something not overly cumbersome, e.g. if I've got a shell open and I want to run "pubtools-quay-(something)" and I want to make it use a certain cert and key, there should be a way I can do that.

Reply:
You're right, during testing I mainly focused on the implementation being properly registered from pub, I didn't pay much attention to the returned value. I decided to use the first returned value because I wanted to make it "best effort"  even though it's debatable whether more than one implementation should even be allowed. If there are no returned values, I decided to use target settings values. I know you're not a big fan of us using target settings, its replacement could be a part of some future improvement.